CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ project(teaserpp VERSION 1.0.0)
 set(CMAKE_CXX_STANDARD 14)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/" ${CMAKE_MODULE_PATH})
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
I'm just curious; is `compile_commands.json` needed to support pip installation?

Reply:
No this is not. The compile flags just are convenient for development. The C++ linters and LSPs can use them to locate included libraries like pybind and Eigen so I don't get a billion errors everywhere.

Reply:
Thanks for your kind answer. I thought `compile_commands.json` was needed pip3. But isn't can be produced by `-DCMAKE_EXPORT_COMPILE_COMMANDS=1`?

Reply:
I think it can yeah. I think when building with `scikit-build` it was easier just to put it in the file. We could probably put a switch there where they aren't made during release builds.